core/src/main/java/com/linecorp/armeria/common/logging/RequestLogLevelMapper.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.logging;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.function.Function;
+
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.annotation.Nullable;
+
+/**
+ * A {@link Function} that determines the {@link LogLevel} of an {@link HttpRequest} from a given
+ * {@link RequestOnlyLog}.
+ *
+ * @see LoggingDecoratorBuilder#requestLogLevelMapper(RequestLogLevelMapper)
+ */
+@FunctionalInterface
+public interface RequestLogLevelMapper extends Function<RequestOnlyLog, LogLevel> {
```

Comment:
`andThen()` and `compose()` in `Function` seem not to match with `XXXLogLevelMapper`. How about removing `Function` from super type?

Reply:
Indeed, thanks for the nice catch.

However, it may be breaking change because it's ambiguous. If we have no choice for breaking, what about removing the old `xxxLogLevelMapper(Function)`?

<img width="945" alt="image" src="https://user-images.githubusercontent.com/15008311/159115450-a3427083-0fbd-411e-9d9b-3427ccb5917b.png">

Reply:
How about overriding `andThen` and `compose` and marking them as deprecated? e.g.
```
/**
 * ...
 * @deprecated Do not use this method.
 */
@Override
@Deprecated
... andThen(...) {
    return Function.super.andThen(...);
}
// Same goes for compose
```